core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -845,6 +852,13 @@ private[spark] object Utils extends Logging {
     customHostname.getOrElse(localIpAddressHostname)
   }
 
+  /**
+   * Get the local machine's URI.
+   */
+  def localHostURI(): String = {
```

Comment:
I don't understand how to use "URI safe" name, something like ```localHostNameURISafe``` or what?
It's good to format URIs only in places where we need it, but currently we have more than one bug because of it.
However we can use something like  ```InetAddresses.toUriString(InetAddress.getByName(Utils.localHostName()))``` to get URI in place without fixing Utils module, but i think that special URI method may prevent URI-based bugs in future due to its existence.